uaclient/cli/cve.py

```diff
@@ -0,0 +1,172 @@
+from collections import namedtuple
+
+from uaclient import messages
+from uaclient.api.u.pro.security.cves.v1 import (
+    CVEVulnerabilitiesOptions,
+    _vulnerabilities,
+)
+from uaclient.api.u.pro.security.fix._common import (
+    query_installed_source_pkg_versions,
+)
+from uaclient.cli import cli_util
+from uaclient.cli.commands import ProArgument, ProArgumentGroup, ProCommand
+from uaclient.cli.formatter import Table
+from uaclient.config import UAConfig
+
+AffectedPackage = namedtuple(
+    "AffectedPackage", ["name", "fix_status", "fix_origin", "fix_version"]
+)
+
+
+@cli_util.with_spinner(msg=messages.CLI_CVES_SPINNER_MSG)
+def _get_cve_vulnerabilities(args, *, cfg: UAConfig, **kwargs):
+    return _vulnerabilities(options=CVEVulnerabilitiesOptions(), cfg=cfg)
+
+
+def _get_affected_pkgs(cve_vulnerabilities, cve_info, cve_name):
+    rows = []
+    installed_pkgs_by_source = query_installed_source_pkg_versions()
+
+    for source_pkg in cve_info.related_packages:
+        binary_pkgs = installed_pkgs_by_source.get(source_pkg, {}).keys()
+
+        for binary_pkg in sorted(binary_pkgs):
+            binary_pkg_info = cve_vulnerabilities.packages.get(binary_pkg)
+
+            if binary_pkg_info:
+                binary_pkg_cve_info = None
+                for cve in binary_pkg_info.cves:
+                    if cve.name == cve_name:
+                        binary_pkg_cve_info = cve
+                        break
+
+                if binary_pkg_cve_info:
+                    rows.append(
+                        AffectedPackage(
+                            name=binary_pkg,
+                            fix_status=binary_pkg_cve_info.fix_status,
+                            fix_origin=binary_pkg_cve_info.fix_origin,
+                            fix_version=binary_pkg_cve_info.fix_version,
+                        )
+                    )
+
+        return rows
+
+
+def _format_affected_pkgs(affected_pkgs):
+    formatted_rows = []
+
+    for affected_pkg in affected_pkgs:
+        if affected_pkg.fix_status == "fixed":
+            formatted_rows.append(
+                [
+                    "<SPACE>{}:".format(affected_pkg.name),
+                    affected_pkg.fix_status,
+                    "({})".format(affected_pkg.fix_origin),
+                    affected_pkg.fix_version,
+                ]
+            )
+        else:
+            formatted_rows.append(
+                [
+                    "<SPACE>{}:".format(affected_pkg.name),
+                    affected_pkg.fix_status,
+                    "",
+                    "",
+                ]
+            )
+
+    return formatted_rows
+
+
+def action_cve(args, *, cfg, **kwargs):
+    cve_vulnerabilities = _get_cve_vulnerabilities(args, cfg=cfg)
+
+    if args.cve not in cve_vulnerabilities.cves:
+        print(messages.CLI_CVE_NOT_AFFECTED.format(issue=args.cve))
+        return
```

Comment:
I think we may still want to show info for CVEs that don't affect the system. But `affected-packages:` will be empty.
This could make things harder :grimacing:

Reply:
But then the API would need to return that information as well and I don't think that's worth it. However, maybe
we display the security page for that CVE in the message. Then the user will at least know where to find more information about it. What do you think ?

Reply:
I think the CLI for `pro cve` cannot rely on the API, because I think it needs to have consistent output for CVEs that don't affect the machine.